src/js/components/FormField/FormField.js

```diff
@@ -597,6 +597,31 @@ const FormField = forwardRef(
     if (typeof required === 'object' && required.indicator === false)
       showRequiredIndicator = false;
 
+    // Check if child is Select or SelectMultiple and modify htmlFor if needed
+    let adjustedHtmlFor = htmlFor;
+    if (htmlFor) {
+      let isSelectComponent = false;
+
+      // Check if children contain Select or SelectMultiple
+      if (children) {
+        Children.forEach(children, (child) => {
+          if (
+            child &&
+            child.type &&
+            (child.type.displayName === 'Select' ||
+              child.type.displayName === 'SelectMultiple')
+          ) {
+            isSelectComponent = true;
+          }
+        });
+      }
+
+      // If it's a Select component and htmlFor doesn't end with __input, add it
+      if (isSelectComponent && !htmlFor.endsWith('__input')) {
```

Comment:
I like the benefit for standard Select/SelectMultiple usage, that it ensures FormField label gets properly associated with the input. Is there any potential concern of people expecting their htmlFor to translate to the DOM exactly as they define it? Is there a risk we'd break something for queries or tests people might already have set up?

I think if we move forward with this, we should clearly define this manipulation in our docs.

@MikeKingdom @britt6612 @jcfilben @halocline

Reply:
I don't think we should be simply checking for something ending in `__input`. We'd want to check for the exact id of the select  + `__input`.

Reply:
we would have to dig into the `child.props` of this `isSelectComponent` we felt like that was getting a little messy and wasnt sure if we need to do that. 

what you mean if FormField has a few children? like this?
```
<Formfield>
<CheckBox/>
<Select/>
</Formfield>
``
???

Reply:
Basically yes. One of the examples for Morpheus was to create a custom formfield that was a textinput for typing in a size plus a select for setting the units (bytes/kbytes...)

Reply:
hmm yeah you got me there.. in that case we would have to look into the childs.props and the `id` and match them.

Reply:
I'm not how they are handling this today. I think they've avoided Grommet so far in a small part due to how they felt it was hard to do custom form fields.

But give `htmlFor` basically makes the browser read that label for a value but also allows clicks on the label to click the related field (in some cases at least, especially checkbox/radiobutton) I'd say in this case it would be set to the text input id. Really for screen reader purposes they'd want to set an aria label that includes the current value plus units.

Reply:
@MikeKingdom okay I added the check to see if the child `id` matches the `htmlFor`